resources/lib/functions.py

```diff
@@ -367,13 +367,16 @@ def addGUIItem(url, details, extraData, folder=True):
     if (len(menuItems) > 0):
         list_item.addContextMenuItems(menuItems, True)
 
+    # new way
+    videoInfoLabels = {}
+
     # add cast
     people = extraData.get('cast')
     if people is not None:
-        list_item.setCast(people)
-
-    # new way
-    videoInfoLabels = {}
+        try:
+            list_item.setCast(people)
```

Comment:
Instead of using a try catch block can we use the version check like what is used for the offscreen ListItem creation check.
I am not a big fan of using exceptions to control logical program flow.